code/modules/chemistry/Reagents-PoisonEtc.dm

```diff
@@ -254,20 +254,35 @@ datum
 			penetrates_skin = 1
 			blob_damage = 5
 			value = 7 // 3 2 1 heat
+			var/counter = 1
 
 			on_mob_life(var/mob/M, var/mult = 1) // -cogwerks. previous version
 				if (!M) M = holder.my_atom
+				if (!counter) counter = 1
 				M.take_toxin_damage(1.5 * mult)
 				if (probmult(8))
 					M.emote("drool")
 				if (prob(10))
 					boutput(M, "<span class='alert'>You cannot breathe!</span>")
-					M.losebreath += (1 * mult)
+					M.losebreath += (mult)
 					M.emote("gasp")
-				if (prob(8))
-					boutput(M, "<span class='alert'>You feel horribly weak.</span>")
-					M.setStatus("stunned", max(M.getStatusDuration("stunned"), 30 * mult))
-					M.take_toxin_damage(2 * mult)
+				switch(counter+= (mult))
+					if (20 to 30)
+						boutput(M, "<span class='alert'>You feel a bit weak.</span>")
+						M.setStatus("stunned", max(M.getStatusDuration("stunned"), 1 SECOND * mult))
+						M.take_toxin_damage(mult)
+						counter = 31
+					if (50 to 60)
+						boutput(M, "<span class='alert'>You feel very weak.</span>")
+						M.setStatus("stunned", max(M.getStatusDuration("stunned"), 2 SECONDS * mult))
```

Comment:
See other comment about what adding `2 SECONDS * mult` stun per `on_mob_life` (where the life loop has `mult` at 2 with no lag) could do. Pretty sure this would permastun.

Reply:
It does skip the step once its done once with the "counter = 61" line so it has to reach the stun step before stunning again, wich would be 15 more cycles. Should i reduce stuns or make the timing between them bigger?

Reply:
Did some testing and it does not permastun, you get stunned for like 3 to 6 seconds and it lasts 10 to 20 seconds before you get stunned again. As said, there is a "counter = 65" wich makes it have to go all the way up to 75 before stunning every time it stuns.

Reply:
Hmmm, I see, but feel like this is overly complicated and will have weird issues in the case of there being some weird `mult` behavior going on. Things that cause that amount of `mult` will probably count as weird behavior anyway (e.g. server melting from someone setting off a whole bunch of big bombs at once), but in any case this reading so differently from the rest of chem code does give me some concern, as the behavior is hard to describe easily.

Reply:
I see... how do you think making something happen every x cycles in a more understandable way could be done? What i did is set a counter that resets when it reaches a threshold and slowly grows back to it.